distribution/lib/Standard/Base/0.0.0-dev/src/Data/Vector.enso

```diff
@@ -1416,6 +1488,207 @@ type Builder
            changes to the builder will not affect the returned vector.
         Vector.from_polyglot_array self.java_builder.toArray
 
+## PRIVATE
+type Build_Style_Builder
+    ## PRIVATE
+
+       A builder type for Enso vectors.
+
+       `Build_Style_Builder` is for use with `Vector.build`, rather than the
+       deprecated `Vector.new_builder`.
+
+       Arguments:
+       - java_builder: The accumulator for the new vector.
+       - warnings: a `Ref` containing `Warning`s that were attached to values
+         added to the builder.
+
+       A vector builder is a mutable data structure, that allows to gather a
+       number of elements and then convert them to a vector. This is
+       particularly useful when the number of elements is not known upfront.
+
+       ! Error Conditions
+
+         - If  an uncaught dataflow error is appended with `append` or
+           `append_vector_range`, an internal panic is thrown, aborting the
+           builder callback that attempted to add the error to the builder. This
+           internal panic is caught by `Vector.build`, which then propagates the
+           dataflow error. The vector is not built.
+
+         - If a dataflow error is thrown from the builder callback, it is
+           propagated by `Vector.build`. The vector is not built.
+
+         - If a value with an attached warning is added to the builder, the
+           vector is built, and the resulting `Vector` keeps the attached
+           warning.
+
+       > Example
+         Append three items.
+
+             Vector.build builder->
+                 builder.append 1
+                 builder.append 2
+                 builder.append 3
+             # => [1, 2, 3]
+    Value java_builder (warnings : Ref)
```

Comment:
```suggestion
    private Value java_builder (warnings : Ref)
```

let's start using Pavel's #9692

I think it's good to not be able to create instances of that builder outside of our factory methods